Code/GraphMol/FileParsers/MolWriters.h

```diff
@@ -76,21 +76,24 @@ class SmilesWriter : public MolWriter {
   //! \brief flush the ostream
   void flush() {
     PRECONDITION(dp_ostream, "no output stream");
-    dp_ostream->flush();
-  };
-
-  //! \brief close our stream (the writer cannot be used again)
-  void close() {
-    PRECONDITION(dp_ostream, "no output stream");
     try {
       dp_ostream->flush();
     } catch (...) {
+      try {
+        dp_ostream->setstate(std::ios::badbit);
+      } catch (const std::runtime_error& e) {
+      }
     }
+  };
+
+  //! \brief close our stream (the writer cannot be used again)
+  void close() {
+    flush();
     std::ostream *tmp_ostream = dp_ostream;
     dp_ostream = NULL;
     if (df_owner) {
-      delete tmp_ostream;
       df_owner = false;
+      delete tmp_ostream;
```

Comment:
Deleting a null pointer is complete legal. What's your concern?

On Saturday, 2 April 2016, DoliathGavid <notifications@github.com> wrote:

> In Code/GraphMol/FileParsers/MolWriters.h
> <https://github.com/rdkit/rdkit/pull/842#discussion_r58294462>:
>
> >        df_owner = false;
> > +      delete tmp_ostream;
>
> similarly, what if dp_ostream was originally a null pointer?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/rdkit/rdkit/pull/842/files/7b5c41b73479400385f9ac859964ac394f0e1232#r58294462>
>